app/controllers/profile_controller.rb

```diff
@@ -89,4 +115,10 @@ def mfa_page_erros
     @secret_code_svg = generate_new_qr(secret_code: @secret_code, email: current_user.email)
     render :show_change_mfa, status: :bad_request
   end
+
+  def refresh_user(updated_user)
+    current_user.given_name = updated_user.user_attributes.find { |attr| attr.name == 'given_name' }.value
+    current_user.family_name = updated_user.user_attributes.find { |attr| attr.name == 'family_name' }.value
+    current_user.email = updated_user.user_attributes.find { |attr| attr.name == 'email' }.value
```

Comment:
we should be updating the current_user.full_name too, given that's what gets displayed. We won't need the `@name = current_user.given_name + " " + current_user.family_name` on line 14 above.

Reply:
actually, we should refactor the full_name construct. It should not be a property on the User model, but rather a method, which takes the current first name and surname. This way we can keep this method as it is (i.e. just updating first name and family name)

Reply:
So all instances where `.full_name` is being used, typically on `@user.full_name`, should just become a helper method like `full_name(current_user.first_name, current_user.last_name)` instead?

Reply:
nope, `full_name` should not be set in here https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/remote_authenticatable.rb#L43 ... but instead, the User model https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/user.rb should have a method called full_name which returns first_name + last_name, similar to what the TeamMember class does: https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/team_member.rb#L13